src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double base1;
+    private double base2;
+    private double height;
+
+    public IsoscelesTrapezoid(Color color, double base1, double base2, double height) {
+        super(color);
+        this.base1 = base1;
+        this.base2 = base2;
+        this.height = height;
+    }
+
+    @Override
+    public double getArea() {
+        return 0.5 * (base1 + base2) * height;
+    }
+
+    @Override
+    public void print() {
+        System.out.println("Figure: isosceles trapezoid, area: " + getArea()
+                + " sq. units, base1: " + base1 + " units, base2: " + base2
+                + " units, height: " + height + " color: " + new ColorSupplier().getRandomColor());

```

Comment:
The use of `new ColorSupplier().getRandomColor()` within the `print()` method is not recommended. Instead, consider using the `color` property inherited from the `Figure` class. This will ensure that the color is consistent and not randomly chosen every time the `print()` method is called.